CONTRIBUTING.md

```diff
@@ -75,6 +82,116 @@ $ bundle exec rake compile -- --with-openssl-dir=$HOME/.openssl/openssl-3.0.1
 $ bundle exec rake test
 ```
 
+#### With OpenSSL master branch
+
+If you want to test, debug, report an issue or contribute to the Ruby OpenSSL
+or [the OpenSSL project](https://www.openssl.org/) including both non-FIPS and
+the [FIPS](https://github.com/openssl/openssl/blob/master/README-FIPS.md) cases,
+you can run the following commands to install the OpenSSL. Testing with the
+latest OpenSSL master branch is a good practice to report an issue to the
+OpenSSL project.
+
+The following configuration options are useful in this case.
+
+* `enable-fips`: Add an option to run with the OpenSSL FIPS module.
+* `enable-trace`: Add an option to enabling tracing log. You can trace logs by
+  implementing a code. See
+  [OSSL_TRACE](https://www.openssl.org/docs/man3.0/man3/OSSL_TRACE.html) for details.
+* compiler flags: You can set debugging compiler flags.
+
+```
+$ git clone https://github.com/openssl/openssl.git
+$ git branch | grep '^*'
+* master
+$ cd openssl
+$ git rev-parse --short HEAD
+0bf18140f4
+$ ./Configure \
+  --prefix=$HOME/.openssl/openssl-fips-debug-0bf18140f4 \
+  --libdir=lib \
+  shared \
+  enable-fips \
+  enable-trace \
+  -O0 -g3 -ggdb3 -gdwarf-5
+$ make -j4
+$ make install
+```
+
+Test the installed OpenSSL.
+
+```
+$ LD_LIBRARY_PATH=$HOME/.openssl/openssl-fips-debug-0bf18140f4/lib \
+  $HOME/.openssl/openssl-fips-debug-0bf18140f4/bin/openssl version
+OpenSSL 3.2.0-alpha3-dev  (Library: OpenSSL 3.2.0-alpha3-dev )
+
+$ LD_LIBRARY_PATH=$HOME/.openssl/openssl-fips-debug-0bf18140f4/lib \
+  $HOME/.openssl/openssl-fips-debug-0bf18140f4/bin/openssl list -providers
+Providers:
+  default
+    name: OpenSSL Default Provider
+    version: 3.2.0
+    status: active
+```
+
+Test the installed OpenSSL with the OpenSSL FIPS config file.
+
+Create a OpenSSL FIPS config `openssl_fips.cnf` file by copying the
+[`openssl_fips.cnf.tmpl`](https://github.com/ruby/openssl/blob/master/test/openssl/fixtures/ssl/openssl_fips.cnf.tmpl)
```

Comment:
By the way, I find this an odd place to put it because it's not used by the test suite itself. At least it's not a test fixture.

I wonder if somewhere like `/tool/openssl_fips.cnf.tmpl` suits better... (though it's not a "tool" either)

Reply:
OK. That makes sense. I will move the file `openssl_fips.cnf.tmpl` to the `tool` directory.

Reply:
I fixed it.